src/App.tsx

```diff
@@ -1,61 +1,130 @@
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const findUser = (userId: number) => (
+  usersFromServer.find(user => user.id === userId) || null
+);
+
+export const newTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: findUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [todos, setTodos] = useState(newTodos);
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [incorrectUser, setIncorrectUser] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+
+  const submitForm = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const hastitleError = !title.trim();
+    const hasIncorrectUser = selectedUser === 0;
+
+    setTitleError(hastitleError);
+    setIncorrectUser(hasIncorrectUser);
+
+    if (hastitleError || hasIncorrectUser) {
+      return;
+    }
+
+    setTodos(prevTodos => {
+      const newId = Math.max(...prevTodos.map(todo => todo.id)) + 1;
+
+      const newTodo = {
+        id: newId,
+        title: title.trim(),
+        completed: false,
+        userId: selectedUser,
+        user: findUser(selectedUser),
+      };
+
+      return [
+        ...prevTodos,
+        newTodo,
+      ];
+    });
+
+    setTitle('');
+    setSelectedUser(0);
+  };
+
+  const onChange = (
+    event:
+    React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    if (event.target.name === 'title') {
+      setTitle(event.target.value);
+      setTitleError(false);
+    }
+
+    if (event.target.name === 'user') {
+      setSelectedUser(+(event.target.value));
+      setIncorrectUser(false);
+    }
+  };
```

Comment:
Create 2 separate function for every handler.
It's bad practice to handle events on 2 different inputs with the same function. What happens if the logic of one of the inputs expands?